schema/proto3/auxiliarygateway.proto

```diff
@@ -0,0 +1,51 @@
+/*
+Copyright 2019 The Alcor Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+        you may not use this file except in compliance with the License.
+        You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+        Unless required by applicable law or agreed to in writing, software
+        distributed under the License is distributed on an "AS IS" BASIS,
+        WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+        See the License for the specific language governing permissions and
+        limitations under the License.
+*/
+
+syntax = "proto3";
+
+package alcor.schema;
+
+option java_package = "com.futurewei.alcor.schema";
+option java_outer_classname = "AuxiliaryGateway";
+
+enum AuxGatewayType {
+    NONE = 0;
+    ZETA = 1;
+}
+
+message AuxGateway {
+    AuxGatewayType AuxGateway_type = 1;
+
+    // if AuxGateway_type == NONE, then id = empty string 
+    string id = 2;
+
+    message destination {
+        string ip_address = 1;
+        string mac_address = 2;
+    }
+
+    // if AuxGateway_type == NONE, then destinations_size = 0 
+    repeated destination destinations = 3;
+
+    message zeta {
+        // port for in-band (same NIC channel) operation
+        uint32 port_inband_operation = 1;
+    }
+
+    oneof extra_info {
```

Comment:
Oneof is the good choice that fits our requirement.

One reminder that it could potentially cause memory crash in C++ code if not used properly, for example, delete one field and then try to set the deleted field. So please watch out for the implementation detail especially on oneof parsing and handling.